go/staking/api/sanity_check.go

```diff
@@ -189,33 +189,30 @@ func (g *Genesis) SanityCheck(now epochtime.EpochTime) error { // nolint: gocycl
 	}
 
 	// All shares of all delegations for a given account must add up to account's Escrow.Active.TotalShares.
-	for acctID, delegations := range g.Delegations {
-		acct := g.Ledger[acctID]
+	for id, delegations := range g.Delegations {
+		acct := g.Ledger[id]
 		if acct == nil {
-			return fmt.Errorf("staking: sanity check failed: delegation specified for a nonexisting account with ID: %v", acctID)
+			return fmt.Errorf("staking: sanity check failed: delegation specified for a nonexisting account with ID: %v", id)
 		}
-		err := SanityCheckDelegations(acct, delegations)
-		if err != nil {
+		if err := SanityCheckDelegations(id, acct, delegations); err != nil {
 			return err
 		}
 	}
 
 	// All shares of all debonding delegations for a given account must add up to account's Escrow.Debonding.TotalShares.
-	for acctID, delegations := range g.DebondingDelegations {
-		acct := g.Ledger[acctID]
+	for id, delegations := range g.DebondingDelegations {
+		acct := g.Ledger[id]
 		if acct == nil {
-			return fmt.Errorf("staking: sanity check failed: debonding delegation specified for a nonexisting account with ID: %v", acctID)
+			return fmt.Errorf("staking: sanity check failed: debonding delegation specified for a nonexisting account with ID: %v", id)
 		}
-		err := SanityCheckDebondingDelegations(acct, delegations)
-		if err != nil {
+		if err := SanityCheckDebondingDelegations(id, acct, delegations); err != nil {
 			return err
 		}
 	}
 
 	// Check the above two invariants for each account as well.
 	for id, acct := range g.Ledger {
-		err := SanityCheckAccountShares(acct, g.Delegations[id], g.DebondingDelegations[id])
-		if err != nil {
+		if err := SanityCheckAccountShares(id, acct, g.Delegations[id], g.DebondingDelegations[id]); err != nil {
```

Comment:
:information_source: some error checking code is changed to this more compact form